iOSClient/Networking/NCAutoUpload.swift

```diff
@@ -210,19 +214,49 @@ class NCAutoUpload: NSObject {
         }
     }
 
-    private func getCameraRollAssets(controller: NCMainTabBarController?, selector: String, alignPhotoLibrary: Bool, account: String, completion: @escaping (_ assets: [PHAsset]?) -> Void) {
-        NCAskAuthorization().askAuthorizationPhotoLibrary(controller: controller) { hasPermission in
+    func processAssets(_ assetCollection: PHAssetCollection, _ fetchOptions: PHFetchOptions, _ tableAccount: tableAccount, _ selector: String, _ account: String) -> [PHAsset] {
+        let assets: PHFetchResult<PHAsset> = PHAsset.fetchAssets(in: assetCollection, options: fetchOptions)
+        var assetResult: [PHAsset] = []
+
+        if selector == NCGlobal.shared.selectorUploadAutoUpload,
+           let idAssets = self.database.getPhotoLibraryIdAsset(image: tableAccount.autoUploadImage, video: tableAccount.autoUploadVideo, account: account) {
+            assets.enumerateObjects { asset, _, _ in
+                var creationDateString = ""
+                if let creationDate = asset.creationDate {
+                    creationDateString = String(describing: creationDate)
+                }
```

Comment:
Isn't this a use case for which optionals are there for?
```suggestion
                var creationDateString: String?

                if let creationDate = asset.creationDate {
                    creationDateString = String(describing: creationDate)
                }
```